app/views/friends/_form.html.erb

```diff
@@ -0,0 +1,42 @@
+<%= form_with(model: friend, local: true) do |form| %>
+  <% if friend.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(friend.errors.count, "error") %> prohibited this friend from being saved:</h2>
+
+      <ul>
+        <% friend.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field form-group">
+   
+  <%= form.text_field :first_name, class:'form-control', placeholder:'First Name' %>
+  </div>
+
+  <div class="field form-group">
+   
+    <%= form.text_field :last_name, class:'form-control', placeholder:'Last Name' %>
+  </div>
+
+  <div class="field form-group">
+   
+    <%= form.text_field :email, class:'form-control', placeholder:'Email' %>
+  </div>
+
+  <div class="field form-group">
+   
+    <%= form.text_field :phone, class:'form-control', placeholder:'Phone' %>
+  </div>
+
+  <div class="field form-group">
+   
+    <%= form.text_field :twitter, class:'form-control', placeholder:'Twitter' %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit class:'btn btn-secondary' %>
+  </div>
+<% end %>
```

Comment:
- **You are supposed to use `form_for` helper and not `form_with` helper.**

Reply:
the form_for is deprecated even in The Odin projects explains it and on the rails guide it shows that the new method is using for_with since rails 5.1 so why are you making us create the form with form_for for a form that at the end of the day we are going to comment out.  The project we presented has full functionality it can create, update, show, destroy it has full functionality so why are we been penalized for creating a finished product?